util/topgen/c.py

```diff
@@ -111,9 +112,6 @@ def __init__(self, top_info, name_to_block: Dict[str, IpBlock]):
         # The .c file needs the .h file's relative path, store it here
         self.header_path = None
 
-        # TODO: Don't hardcode the address space used for software.
-        self.addr_space = "hart"
-
```

Comment:
I think there needs to be an address space selected somewhere, but it could be provided as an argument to the generator object's constructor. The header file generated is _not_ intended to spit out macros for more than one address space at a time.

We could possibly consider doing that, but then the macro names would need to start specifying their address spaces, and it's a lot of irrelevant info for the compilation target, right?

Or does not having the address space in the macro name end up causing issues when one CPU actually connects to one node's multiple address spaces through translation layers? (In the SoC context, the CPU might need multiple headers, one for each address space, and that might be fine only if there aren't any name clashes.)